### PR TITLE
Support timestamp(n) SQL type

### PR DESCRIPTION
**Pull request by findepi**
Relates to https://github.com/apache/datafusion/discussions/13230

### CODE REVIEW

File: datafusion/sql/src/planner.rs [hunk 1]
```diff
@@ -466,7 +467,14 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
                     // Timestamp Without Time zone
                     None
                 };
-                Ok(DataType::Timestamp(TimeUnit::Nanosecond, tz.map(Into::into)))
+                let precision = match precision {
```
Comment (caicancai) [inline datafusion/sql/src/planner.rs:470]: Can I assume this is the standard between vector api and presion?

Comment (caicancai) [reply datafusion/sql/src/planner.rs:470]: I think the relationship between vector api and precision can be briefly described

File: datafusion/sqllogictest/test_files/timestamps.slt [hunk 2]
```diff
@@ -402,6 +402,41 @@ SELECT COUNT(*) FROM ts_data_secs where ts > from_unixtime(1599566400)
...
 
+query P rowsort
+SELECT ts FROM ts_data_nanos;
+----
+2020-09-08T11:42:29.190855123
+2020-09-08T12:42:29.190855123
+2020-09-08T13:42:29.190855123
+
+query P rowsort
+SELECT CAST(ts AS timestamp(0)) FROM ts_data_nanos;
```
Comment (alamb) [inline datafusion/sqllogictest/test_files/timestamps.slt:413]: This is very cool 👍

**Review by alamb — APPROVED** [review]
This looks great to me -- thank you @findepi  -- it will be amazing to be able to specify the timestamp precision via SQL

**Review by caicancai — APPROVED** [review]

File: datafusion/sqllogictest/test_files/timestamps.slt (hunk 2 above)
Comment (caicancai) [reply datafusion/sqllogictest/test_files/timestamps.slt:413]: The Calcite Arrow adapter will also comply with this standard

**Comment by alamb** [comment]
Thanks again @findepi

**Pull request merged**